src/app/services/challengesApi.ts

```diff
@@ -0,0 +1,18 @@
+import { api } from './api';
+import challengeDataType from '@/interfaces/challenge.type';
+
+type ChallengesQueryResponse = { message: string; challenges: challengeDataType[] };
+
+export const challengesApi = api.injectEndpoints({
+  endpoints: (builder) => ({
+    getChallenges: builder.query<ChallengesQueryResponse, void>({
+      query: () => '/challenges',
+      providesTags: ['Challenges'],
+      transformResponse: (response: ChallengesQueryResponse) => {
```

Comment:
I can't find the sense of transformResponse here if you are sending the response as it is